Geometry/HGCalCommonData/src/HGCalDDDConstants.cc

```diff
@@ -387,14 +387,14 @@ int HGCalDDDConstants::getLayer(double z, bool reco) const {
   double zz =
       (reco ? std::abs(z) : HGCalParameters::k_ScaleFromDDD * std::abs(z));
   const auto& zLayerHex = hgpar_->zLayerHex_;
-  std::find_if(zLayerHex.begin() + 1, zLayerHex.end(),
-               [&k, &zz, &zLayerHex](double zLayer) {
-                 ++k;
-                 return zz < 0.5 * (zLayerHex[k - 1] + zLayerHex[k]);
-               });
-  int lay = k;
+  auto itr = std::find_if(zLayerHex.begin() + 1, zLayerHex.end(),
+			  [&k, &zz, &zLayerHex](double zLayer) {
+			    ++k;
+			    return zz < 0.5 * (zLayerHex[k - 1] + zLayerHex[k]);
+			  });
+  int lay =(itr == zLayerHex.end()) ? (int)(zLayerHex.size()) :  k;
```

Comment:
```suggestion
  int lay =(itr == zLayerHex.end()) ? static_cast<int>(zLayerHex.size()) :  k;
```

Reply:
C++ style casts are preferred by the coding [rules](https://github.com/cms-sw/cms-sw.github.io/blob/3e89200bde531a9aa1d01d4c92c1d65a5f6c87fd/cms_coding_rules.md#4----technical-coding-rules-1).

OK to make this change in a later PR.

Reply:
Done